### DISCUSSION TITLE
Can't check if Popup exists or not

### DISCUSSION THREAD

**Discussion by Muhammadehsan2019** [cypress-io/cypress#15194] [Questions and Help]
The Scenario is that I want to make sure to click on the Popup close button if it exists.
the popup is dynamic as sometimes it exists but sometimes not, and cypress can't automatically handle it because its different from the  alert. the below code will only work if the popup exist, and it will fail when poput is not there.
"Expected to find element:......but never found it. "

  // Click "Close" if the SOP popup is visible.
         actionDetailsObj.getCloseSOPPopUpButton().then((SOPPopUpButton) => {
            if (SOPPopUpButton.is(':visible')) {
               actionDetailsObj.getCloseSOPPopUpButton().click()
            }
         })

**Comment by herecydev** [comment]
This sounds like something that needs reworking so that you can reliably put it into that state, otherwise it will be flaky.

Theres lots of info here that should help: https://docs.cypress.io/guides/core-concepts/conditional-testing.html
